### ISSUE TITLE
Error running JFlex Generator

### ISSUE THREAD

**Issue by akrylysov** [JetBrains/Grammar-Kit#65]
When I click "Run JFlex Generator" I get error message:
```
/System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Home/bin/java -Xmx512m -Dfile.encoding=UTF-8 -classpath /Users/artem/Work/idea-nsis/JFlex.jar JFlex.Main -sliceandcharat -skel /Users/artem/Work/idea-nsis/idea-flex.skeleton -d /Users/artem/Work/idea-nsis/src/com/krylysov/nsisplugin /Users/artem/Work/idea-nsis/src/com/krylysov/nsisplugin/Nsis.flex
Exception in thread "main" java.lang.UnsupportedClassVersionError: JFlex/Main : Unsupported major.minor version 52.0
```

I tried to set JDK 1.7 and JDK 1.8 for the project, but it didn't help.

**Comment by nicity** [comment]
As work around download IntelliJ IDEA with bundled jdk 8

28 Мар 2015 г. 21:22 пользователь "Artem Krylysov" <notifications@github.com>
написал:
>
> When I click "Run JFlex Generator" I get error message:
>
> /System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Home/bin/java
-Xmx512m -Dfile.encoding=UTF-8 -classpath
/Users/artem/Work/idea-nsis/JFlex.jar JFlex.Main -sliceandcharat -skel
/Users/artem/Work/idea-nsis/idea-flex.skeleton -d
/Users/artem/Work/idea-nsis/src/com/krylysov/nsisplugin
/Users/artem/Work/idea-nsis/src/com/krylysov/nsisplugin/Nsis.flex
> Exception in thread "main" java.lang.UnsupportedClassVersionError:
JFlex/Main : Unsupported major.minor version 52.0

**Comment by EBatTiVo** [comment]
This is a message from Java.  You are using a JRE that is older than the Java language version that the JFlex Generator was built for.  Either re-build the plugin on your own machine, or upgrade your JRE/JDK.  Java 1.6 (which you -- and a lot of other people -- are still running) has reached it's end-of-life and is no longer updated or available from Oracle.

If you need to rebuild using a newer JDK and still need to target 1.6 level JREs, then add '-target 1.6' to your build command line.  See http://docs.oracle.com/javase/7/docs/technotes/tools/windows/javac.html

**Comment by akrylysov** [comment]
It is possible to get JDK from the project configuration, see my proposed fix https://github.com/JetBrains/Grammar-Kit/pull/66.

**Comment by nicity** [comment]
We rebuilt jflex.jar with java 6

It is possible to get JDK from the project configuration, see my proposed
fix #66 <https://github.com/JetBrains/Grammar-Kit/pull/66>.

—
Reply to this email directly or view it on GitHub
<https://github.com/JetBrains/Grammar-Kit/issues/65#issuecomment-87341893>.
